crates/cli/commands/src/import_era.rs

```diff
@@ -1,25 +1,71 @@
-//! Command that initializes the node by importing a chain from a file.
+//! Command that initializes the node by importing a chain from ERA files.
 use crate::common::{AccessRights, CliNodeTypes, Environment, EnvironmentArgs};
-use clap::Parser;
+use clap::{Args, Parser, ValueEnum};
+use eyre::OptionExt;
+use reqwest::{Client, Url};
 use reth_chainspec::{EthChainSpec, EthereumHardforks};
 use reth_cli::chainspec::ChainSpecParser;
-use reth_era_downloader::read_dir;
+use reth_era_downloader::{read_dir, EraClient, EraStream, EraStreamConfig};
+use reth_era_utils as era;
 use reth_etl::Collector;
-use reth_node_core::version::SHORT_VERSION;
+use reth_node_core::{dirs::data_dir, version::SHORT_VERSION};
 use std::{path::PathBuf, sync::Arc};
 use tracing::info;
 
-/// Syncs ERA encoded blocks from a directory.
+/// Syncs ERA encoded blocks from a local or remote source.
 #[derive(Debug, Parser)]
 pub struct ImportEraCommand<C: ChainSpecParser> {
     #[command(flatten)]
     env: EnvironmentArgs<C>,
 
+    #[clap(flatten)]
+    import: ImportArgs,
+}
+
+#[derive(Debug, Args)]
+#[group(required = true, multiple = false)]
+pub struct ImportArgs {
     /// The path to a directory for import.
     ///
-    /// The ERA1 files are read from the directory parsing headers and bodies.
-    #[arg(value_name = "IMPORT_PATH", verbatim_doc_comment)]
-    path: PathBuf,
+    /// The ERA1 files are read from the local directory parsing headers and bodies.
+    #[arg(value_name = "IMPORT_ERA_PATH", verbatim_doc_comment)]
+    path: Option<PathBuf>,
+
+    /// The URL to a remote host where the ERA1 files are hosted.
+    ///
+    /// The ERA1 files are read from the remote host using HTTP GET requests parsing headers
+    /// and bodies.
+    #[arg(value_name = "IMPORT_ERA_URL", verbatim_doc_comment)]
+    url: Option<Url>,
+
+    /// The chain for which a known URL exists.
+    ///
+    /// When specified, the URL is derived from the chain name and read from the remote host using
+    /// HTTP GET requests parsing headers and bodies.
+    #[arg(value_enum, value_name = "IMPORT_ERA_CHAIN", verbatim_doc_comment)]
+    chain_id: Option<Chain>,
```

Comment:
this should also `conflicts_with = "url"`?

Reply:
This is covered by `#[group(required = false, multiple = false)]` on the struct.

If you specify both, then it prints error like this:
```
error: the argument '--path <IMPORT_ERA_PATH>' cannot be used with '--url <IMPORT_ERA_URL>'
```

Reply:
ah nice